closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,12 +562,257 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+    num_of_generic = 4;
+#elif defined(__linux__)
+    num_of_generic = 1;
+#endif
+
+    /* Library names for OpenSSL 1.1.1, 1.1.0 and symbolic links
+     * It is important to preserve the order!!!
+     */
+    static const char * const libNames[] = {
+#if defined(_AIX)
+        "libcrypto.a(libcrypto64.so)",      /* general symlink library name from archive file */
+        "libcrypto64.so",                   /* general symlink library name */
+        "libcrypto.a(libcrypto.so)",        /* general symlink library name from archive file */
+        "libcrypto.so",                     /* general symlink library name */
+        "libcrypto.a(libcrypto64.so.3)",    /* 3.x library name from archive file */
+        "libcrypto64.so.3",                 /* 3.x library name */
+        "libcrypto.a(libcrypto.so.3)",      /* 3.x library name from archive file */
+        "libcrypto.so.3",                   /* 3.x library name */
+        "libcrypto.a(libcrypto64.so.1.1)",  /* 1.1.x library name from archive file */
+        "libcrypto.so.1.1",                 /* 1.1.x library name */
+        "libcrypto.so.1.0.0",               /* 1.0.x library name */
+#elif defined(__APPLE__)
+        "libcrypto.3.dylib",                /* 3.x library name */
+        "libcrypto.1.1.dylib",              /* 1.1.x library name */
+        "libcrypto.1.0.0.dylib",            /* 1.0.x library name */
+        /* "libcrypto.dylib"                Apple no longer supports loading symlink. */
+#elif defined (_WIN32)
+        "libcrypto-3-x64.dll",              /* 3.x library name */
+        "libcrypto-1_1-x64.dll",            /* 1.1.x library name */
+        "libeay32.dll",                     /* old library name */
+#else
+        "libcrypto.so",                     /* general symlink library name */
+        "libcrypto.so.3",                   /* 3.x library name */
+        "libcrypto.so.1.1",                 /* 1.1.x library name */
+        "libcrypto.so.1.0.0",               /* 1.0.x library name */
+        "libcrypto.so.10",                  /* old library name */
+#endif
+    };
+
+    numOfLibs = (sizeof(libNames) / sizeof(libNames[0]));
+    /* If JAVA_HOME is not null or empty and no library has been loaded yet, try there. */
+    if ((NULL != chomepath) && ('\0' != *chomepath) && (NULL == crypto_library)) {
+        char **libNamesWithPath = malloc(size * sizeof(char *));
+        size_t path_len = strlen(chomepath);
+        char *libPath = malloc((path_len + 16) * sizeof(char));
+        strcpy(libPath, chomepath);
+        /* Append the proper directory using a slash or backslash, depending on the operating system. */
+#if defined(_WIN32)
+        strcat(libPath, "\\bin\\");
+#else /* defined(_WIN32) */
+        strcat(libPath, "/lib/");
+#endif /* defined(_WIN32) */
+        if (traceEnabled) {
+            fprintf(stdout, "Attempting to load library bundled with JDK from: %s\n", libPath);
+        }
+
+        for (; i < numOfLibs; i++) {
```

Comment:
Please include the initializer, even if it's redundant.
```c
        for (i = 0; i < numOfLibs; i++) {
```

Reply:
Included.